packages/checkout/sdk/src/smartCheckout/balanceCheck/balanceRequirement.ts

```diff
@@ -141,10 +160,22 @@ export const getTokenBalanceRequirement = (
       },
       current: {
         ...tokenBalanceResult,
+        token: {
+          address: tokenBalanceResult.token.address,
+          symbol,
+          name,
+          decimals,
+        },
```

Comment:
@dreamoftrees `tokenBakanceResult` is of type `TokenBalance` 

```
export type TokenBalance = {
  /** Type to indicate this is a native or ERC20 token. */
  type: ItemType.NATIVE | ItemType.ERC20;
  /** The balance of the item. */
  balance: BigNumber;
  /** The formatted balance of the item. */
  formattedBalance: string;
  /**  The token info of the item. */
  token: TokenInfo;
};
```

do we need to to have `token: {...}` as well?